src/cli/cli.cpp

```diff
@@ -4570,21 +4600,18 @@ void Interpreter::ProcessLine(char *aBuf)
 
     OT_ASSERT(aBuf != nullptr);
 
+    // Ignore the command if another command is pending.
+    VerifyOrExit(!mCommandIsPending, args[0].Clear());
```

Comment:
Can we instead of doing `args[0].Clear()` set the error to `OT_ERROR_PENDING`?
I think we handle it the same way in `exit` but may be more clear to set the error. Thoughts?

Reply:
If we set the error to `OT_ERROR_PENDING` here, it will `OutputResult(OT_ERROR_PENDING) at exit, which is not what we want. 

The logic here is a little tricky. There are 4 cases we want to handle:
1. If the command executed sucessfully, we should `OutputResult(error)` with error being `OT_ERROR_NONE`.
2. If the command execution failed, we should `OutputResult(error)` with error not being `OT_ERROR_NONE`. 
3. If there is another command executing, we should output nothing by just ignoreing the input
4. If the input is empty, and there is no other command executing, we should only output a new prompt

So, we are implementing case 3 by ignore the input if another command is executing.

Reply:
I looked at `OutputResult(OT_ERROR_PENDING)` again and found that it actually does nothing. 

So, you are correct that setting error to `OT_ERROR_PENDING` can be effectively same. But by doing so, we don't know at exit if CLI is executing the current command input or there is another command executing and the current command input is ignored. I am worried the unclearance may cause other issues in the future. 

Currently, the conditions are clear:
- If `error != OT_ERROR_NONE` , the current command was executed and error was returned
- If `error == OT_ERROR_NONE && !args[0].IsEmpty()`, the current command was executed successfully
- If `error == OT_ERROR_NONE && args[0].IsEmpty()`, the current command was ignored either because it's empty or because there is already an outstanding command executing.

Reply:
Thanks. SGTM.